lib/logging/lib/logging.c

```diff
@@ -109,29 +109,29 @@ log_impl(
 
         merr_strerror(err, buf, sizeof(buf));
 
-        rc = snprintf(scratch_buffer, sizeof(scratch_buffer), "[HSE] %s:%d %s: %s: %s",
-            ev->ev_file, ev->ev_line, ev->ev_dte.dte_func, fmt, buf);
+        rc = snprintf(scratch_buffer, sizeof(scratch_buffer), "[HSE] %s:%d %s: %s: %s%s",
+            ev->ev_file, ev->ev_line, ev->ev_dte.dte_func, fmt, buf, log_file ? "\n" : "");
```

Comment:
didn't really see a cleaner way to do this unless syslog strips ending new lines

Reply:
From the man page:  The remaining arguments are a format, as in printf(3), and any arguments required by the format, except that the two-character sequence %m will be replaced by the error message string strerror(errno).  The format 
 string need not include a terminating newline character.

Reply:
So you could just put a \n in the format I think...

Reply:
I would replace scratch_buffer with hse_logbuf_tls, which tells me it came from hse, is used for logging, and is in thread-local-storage....

Reply:
how often are you looking at global symbols?

Reply:
the \n thing was a good call